x-pack/plugins/aiops/public/components/spike_analysis_table/use_view_in_log_pattern_analysis_action.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+
+import { SerializableRecord } from '@kbn/utility-types';
+import { fromKueryExpression, toElasticsearchQuery } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import type { SignificantTerm } from '@kbn/ml-agg-utils';
+
+import { SEARCH_QUERY_LANGUAGE } from '../../application/utils/search_utils';
+import { useAiopsAppContext } from '../../hooks/use_aiops_app_context';
+
+import { TableActionButton } from './table_action_button';
+import { getTableItemAsKQL } from './get_table_item_as_kql';
+import type { GroupTableItem, TableItemAction } from './types';
+
+const viewInLogPatternAnalysisMessage = i18n.translate(
+  'xpack.aiops.spikeAnalysisTable.linksMenu.viewInLogPatternAnalysis',
+  {
+    defaultMessage: 'View in Log Pattern Analysis',
+  }
+);
+
+export const useViewInLogPatternAnalysisAction = (dataViewId?: string): TableItemAction => {
+  const { application, share, data } = useAiopsAppContext();
+
+  const mlLocator = useMemo(() => share.url.locators.get('ML_APP_LOCATOR'), [share.url.locators]);
+
+  const generateLogPatternAnalysisUrl = async (
```

Comment:
Now that browsers can generally handle URLs 32k more in length, I guess we don't need to worry too much about how long the URL that can be generated here could be? I got up to 5000 chars for a group of 23 items.